packages/ui/src/ui/components/Tabs/Tabs.tsx

```diff
@@ -89,8 +90,13 @@ class Tabs<ValueT extends string = string> extends React.Component<Props<ValueT>
     renderLink(item: TabItem<ValueT>) {
         const {active, onTabChange} = this.props;
         const onTabClick = (evt: React.MouseEvent<Element, MouseEvent>) => {
-            const clickHandler = action.makeEntryClickHandler(evt, onTabChange);
-            clickHandler(item.value);
+            const currentItem = this.getCurrentItem();
```

Comment:
`const currentItem = this.getCurrentItem();`
`if (!currentItem?.canLeaveTab || currentItem.canLeaveTab()) {`
`    const clickHandler = action.makeEntryClickHandler(evt, onTabChange);`
`    clickHandler(item.value);`
`}`

Reply:
Fixed